provider_packages/SETUP_TEMPLATE.py.jinja2

```diff
@@ -0,0 +1,105 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# NOTE! THIS FILE IS AUTOMATICALLY GENERATED AND WILL BE\
+# OVERWRITTEN WHEN RUNNING
+#
+# ./breeze prepare-provider-readme
+#
+# IF YOU WANT TO MODIFY IT, YOU SHOULD MODIFY THE TEMPLATE
+# `SETUP_TEMPLATE.py.jinja2` IN the `provider_packages` DIRECTORY
+
+"""Setup.py for the {{ PACKAGE_PIP_NAME }} package."""
+
+import io
+import logging
+import os
+import sys
+
+from os.path import dirname
+from setuptools import find_packages, setup
+
+logger = logging.getLogger(__name__)
+
+version = '{{ RELEASE_NO_LEADING_ZEROS }}{{VERSION_SUFFIX }}'
+
+my_dir = dirname(__file__)
+
+try:
+    with io.open(os.path.join(my_dir,
+                              '{{ PROVIDER_PATH }}/{{ README_FILE }}'),
+                 encoding='utf-8') as f:
+        long_description = f.read()
+except FileNotFoundError:
+    long_description = ''
+
+
+def do_setup(version_suffix_for_pypi=''):
+    """
+    Perform the package setup.
+    """
+    setup(
+        name='{{ PACKAGE_PIP_NAME }}',
+        description='{{ PROVIDER_TYPE }} package {{ PACKAGE_PIP_NAME }} for Apache Airflow',
+        long_description=long_description,
+        long_description_content_type='text/markdown',
+        license='Apache License 2.0',
+        version=version + version_suffix_for_pypi,
+        packages=find_packages(
+            include=['airflow.providers.{{ PROVIDER_PACKAGE_ID }}*']),
+        zip_safe=False,
+        install_requires={{ INSTALL_REQUIREMENTS }},
+        setup_requires={{ SETUP_REQUIREMENTS }},
+        extras_require={{ EXTRAS_REQUIREMENTS }},
+        classifiers=[
+            'Development Status :: 5 - Production/Stable',
+            'Environment :: Console',
+            'Environment :: Web Environment',
+            'Intended Audience :: Developers',
+            'Intended Audience :: System Administrators',
+            'License :: OSI Approved :: Apache Software License',
+            'Programming Language :: Python :: 3.6',
+            'Programming Language :: Python :: 3.7',
+            'Programming Language :: Python :: 3.8',
+            'Topic :: System :: Monitoring',
+        ],
+        author='Apache Software Foundation',
+        author_email='dev@airflow.apache.org',
+        url='http://airflow.apache.org/',
+        download_url=(
+            'https://dist.apache.org/repos/dist/release/airflow/{{ PROVIDERS_FOLDER }}' + version),
+        python_requires='~=3.6',
+    )
+
+#
+# Note that --version-suffix-for-pypi should only be used in case we generate RC packages for PyPI
+# Those packages should have actual RC version in order to be published even if source version
+# should be the final one.
+#
+if __name__ == "__main__":
+    suffix=''
+    if len(sys.argv) == 1:
+        print("ERROR! Missing first param", file=sys.stderr)
```

Comment:
This line is the problem - it stops setup.py being used normally. Please make it possible to ruin without passing any args

Reply:
I still don't understand why we need this - we don't commit the files, so when releasing rc to pypi why not just regenerate this file with the rc already in the version?

Reply:
Yep the line is not needed.

My intention (as I explained when submitted this PR) IS to commit those files. This way (as I explained there) it will be easier to track changes (for example in the dependencies) for those packages.

Reply:
But If you have good reasons, why we should not commit those files, I am happy to discuss and hear the reasoning . I basically followed the same pattern when I always generate and commit README files when I run "Prepare" step - mainly because this is the time I can dump the release history. But for setup it is pretty much convenience only and I could potentially move it to the package generation time. 

Let me know what you think!

Reply:
Nope, happy to commit them - just assumed cos we have a template file and you ignored them they we wouldn't be commiting them

Reply:
I hope I only ignored the "provider_package" ones :)

Reply:
Oh yeah. Only the ones from "provider_packages" :D. There whole `airflow` tree is ignored as well as all those is dynamically overwritten during packaging.